integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/checker/DataQualityChecker.java

```diff
@@ -20,96 +20,398 @@
+            }
+        }
+    }
+
+    /**
+     * Main method that conducts the data quality check. This is done by getting row counts and
+     * iterate through each data quality check. The results summary is provided after all are
+     * executed. The results are written to log as well as hive
+     */
+    protected boolean doDataQualityChecks() {
+
+        boolean isSuccessful = false;
+        try {
+            String databaseName = flowAttributes.getAttributeValue(DataQualityConstants.CATEGORY_ATTRIBUTE);
+            String tableName = flowAttributes.getAttributeValue(DataQualityConstants.FEED_ATTRIBUTE);
+            String processing_dttm = flowAttributes.getAttributeValue(DataQualityConstants.PROCESSING_DTTM_ATTRIBUTE);
+
+            String feedTableName = tableName + DataQualityConstants.FEED_TABLE_SUFFIX;
+            String invalidTableName = tableName + DataQualityConstants.INVALID_TABLE_SUFFIX;
+            String validTableName = tableName + DataQualityConstants.VALID_TABLE_SUFFIX;
+            String whereClause = DataQualityConstants.PROCESSING_DTTM_COLUMN + " = '" + processing_dttm + "'";
+
+            setSourceRowCount();
+
+            long rowCount;
+            rowCount = getRowCount(databaseName, feedTableName, whereClause);
+            flowAttributes.addAttribute(DataQualityConstants.DQ_FEED_ROW_COUNT_ATTRIBUTE, String.valueOf(rowCount));
+
+            rowCount = getRowCount(databaseName, invalidTableName, whereClause);
+            flowAttributes.addAttribute(DataQualityConstants.DQ_INVALID_ROW_COUNT_ATTRIBUTE, String.valueOf(rowCount));
+
+            rowCount = getRowCount(databaseName, validTableName, whereClause);
+            flowAttributes.addAttribute(DataQualityConstants.DQ_VALID_ROW_COUNT_ATTRIBUTE, String.valueOf(rowCount));
+
+            // Execute Data Quality rules
+            boolean rulePass;
+            boolean dqAllRulePass = true;
+            for (DataQualityRule rule : ruleList) {
+
+                log.info("Executing rule: " + rule.getName());
+
+                if (rule.loadAttributes(flowAttributes)) {
+                    rulePass = rule.evaluate();
+                } else {
+                    rulePass = false;
+                }
+
+                if (!rulePass) {
+                    log.error("Rule: " + rule.getName() + " has FAILED. Rule desc: " + rule.getDescription());
+                } else {
+                    log.info("Rule: " + rule.getName() + " has PASSED");
+                }
+
+                dqAllRulePass = dqAllRulePass && rulePass;
+
+            }
+
+            outputToHive();
+
+            outputToLog();
+
+            // Return false if any rule failed
+            if (dqAllRulePass == false) {
+                log.warn("Data Quality check failures");
+                isSuccessful = false;
+            } else {
+                isSuccessful = true;
+            }
+
+        } catch (MissingAttributeException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)